backend/tests/Tests/Serialization.DarkTypes.Tests.fs

```diff
@@ -0,0 +1,285 @@
+module Tests.DarkTypesSerialization
+
+open Expecto
+open System.Text.RegularExpressions
+
+open Prelude
+open Tablecloth
+open TestUtils.TestUtils
+
+
+module File = LibBackend.File
+module Config = LibBackend.Config
+
+module PT = LibExecution.ProgramTypes
+module RT = LibExecution.RuntimeTypes
+
+module PT2DT = StdLibDarkInternal.Helpers.ProgramTypesToDarkTypes
+
+module PT2RT = LibExecution.ProgramTypesToRuntimeTypes
+
+
+module V = SerializationTestValues
+
+
+
+module RoundtripTests =
+  // for each type/value:
+  // perform internal -> DT -> internal -> DT -> internal
+  // most of the time, it should end up being the same as the source.
+  // if there are known exceptions, break down individual mappings as separate tests
+
+  let testRoundtrip
+    (testName : string)
+    (typeName : RT.TypeName.T)
+    (original : 'a)
+    (toDT : 'a -> RT.Dval)
+    (fromDT : RT.Dval -> 'a)
+    =
+    testTask testName {
+      let firstDT = original |> toDT
+
+      let context =
+        LibExecution.TypeChecker.Context.FunctionCallResult(
+          fnName =
+            (RT.FnName.fqPackage
+              "Darklang"
+              (NonEmptyList.ofList [ "LanguageTools"; "ProgramTypes" ])
```

Comment:
We can replace "Darklang", "LanguageTools", "ProgramTypes", and "expr" with anything and nothing breaks. If I remember correctly, context only becomes relevant when tests fail, but I am not sure what the right thing to do here